src/Toml/Printer.hs

```diff
@@ -134,36 +127,54 @@ prettyKeyValue options i =
             . formatTime defaultTimeLocale "%z"
 
 -- | Returns pretty formatted tables section of the 'TOML'.
-prettyTables :: PrintOptions -> Int -> Text -> PrefixMap TOML -> Text
-prettyTables options i pref =
-    Text.concat . map (prettyTable . snd) . orderWith options . HashMap.toList
+prettyTables :: PrintOptions -> Int -> Text -> PrefixMap TOML -> [Text]
+prettyTables options i pref = mapOrdered (prettyTable . snd) options
   where
-    prettyTable :: PrefixTree TOML -> Text
+    prettyTable :: PrefixTree TOML -> [Text]
     prettyTable (Leaf k toml) =
-        let name = getPref k in mconcat
-            [ tabWith options i
-            , prettyTableName name
-            , prettyTomlInd options (succ i) name toml ]
+        let name = addPrefix k pref
+        in "": tabWith options i <> prettyTableName name :
+             dropWhile (== "") (prettyTomlInd options (succ i) name toml)
```

Comment:
This addition of `""` and `dropWhile (== "")` looks very weird... At this point I think that using `T.intercalate` is better than adding strange logic :thinking: